### PR TITLE
Initial TenX Support

### PR DESCRIPTION
**Pull request by saurabsa**
- Added tenx command under 'container' sub-group

- Checks for 'TENX_HOME' and executes the core clr command 'tenx.dll'

- Parameter 'args' accepts argument to pass to tenx.dll

### CODE REVIEW

File: src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py [hunk 1]
```diff
@@ -404,3 +405,67 @@ def _get_service_token():
...
+    # os.system("SETX {0} {1} /M".format(env_var, env_val))
+    """
+
+    file_path = os.environ[env_var]
+    return file_path
+
+def _install_tenx(npm_dir):
+    try:
+        encoding = "utf-8"
+        if platform.system() == 'Windows':
```
Comment (philon-msft) [inline src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py]: Could use ternary operator here too

File: src/command_modules/azure-cli-container/azure/cli/command_modules/container/_help.py [hunk 2]
```diff
@@ -19,3 +19,8 @@
     type: group
     short-summary: "Set up automated builds for a multi-container Docker application."
 """
+
+helps['container tenx'] = """
```
Comment (saurabsa) [inline src/command_modules/azure-cli-container/azure/cli/command_modules/container/_help.py]: Yes we will update it later to a better name. I guess tenx command with be replaced by the arguments in tenx dll like connect and run. Will revisit this then again.

---
In reply to: [103980747](https://github.com/saurabsa/azure-cli/pull/1#discussion_r103980747) [](ancestors = 103980747)

File: src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py [hunk 3]
```diff
@@ -404,3 +405,67 @@ def _get_service_token():
...
+
+        if p.returncode != 0:
+            print ('TENX_HOME should be set correctly in the PATH')
+            raise CalledProcessError(p.returncode, cmd)
+
+    except Exception as inst:
+        print(inst)
+
+def _ensure_tenx():
+    env_var = "TENX_HOME"
```
Comment (saurabsa) [inline src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py]: for that, we would have to update the env name in tenx.dll, which I am not sure, we want to do. We can re-visit this.

---
In reply to: [103827293](https://github.com/saurabsa/azure-cli/pull/1#discussion_r103827293) [](ancestors = 103827293)

File: src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py [hunk 4]
```diff
@@ -404,3 +405,65 @@ def _get_service_token():
...
+            raise CalledProcessError(p.returncode, cmd)
+    except Exception as e:
+        raise CLIError(e)
+
+def _ensure_tenx():
+    try:
+        env_var = "TENX_HOME"
+
+        # TODO: Uncomment below code when we install tenx at a specific location
+        """
```
Comment (PeterJausovec) [inline src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py]: remove

**Pull request closed without merging**